### DISCUSSION TITLE
Issue with ‘Get_member’ response in RFEM 6.06.0004 and later and the flag “I” in results response in RFEM 6.06.0002 and later

### DISCUSSION THREAD

**Discussion by sonkaste1** [Dlubal-Software/Dlubal_CSharp_Client#59] [Q&A] [answered]
Hello there!

we have 2 questions/issues with RFEM 6 API, one relates to location flag "M" vs "M I I" which is not clear to us. 
The second relates to "Get_member" where we receive "0" value for section_end and section_start which is UNEXPECTED.

**1) Issue with the flag “I” in results response in RFEM 6.06.0002 and higher**
-	During reading the results from RFEM API, we are checking the property location_flags to decide, if we will use this result or not
-	We found out, that there is a difference between response in RFEM 6.06.0001 and RFEM 6.06.0002(+ higher)
-	In 6.06.0002 and higher, it seems that there is now a flag “I” used for almost every result we read. But in older (6.06.0001) the flag “I” was not used there that much.

-	Why is that? What is the exact meaning of this “I” flag? 
-	Here is example, but it is related to lot of other results:
![image](https://github.com/Dlubal-Software/Dlubal_CSharp_Client/assets/87471266/7e122d22-982e-4ed5-8729-d6766bc48bde)

-	Attached the are XML files as responses from RFEM API – we used Boomerang tool for all of those
-	I checked it in Visual Studio during the debugging and there are the same responses as in Boomerang this time

**2) Issue with ‘Get_member’ response in RFEM 6.06.0004**
-	We are using the ‘get_member(int no)’ function to get member data .
-	In response the section_start and section_end are set to “0” when we use RFEM 6.06.0004 and later.
-	Here are the screenshots from Visual Studio during debugging (Nuget used RFEM 6.5.5)

![image](https://github.com/Dlubal-Software/Dlubal_CSharp_Client/assets/87471266/5716fa62-8156-4507-beeb-2edcc4523380)

![image](https://github.com/Dlubal-Software/Dlubal_CSharp_Client/assets/87471266/56e85d0c-8d43-4233-854e-561b2bad6cb9)

![image](https://github.com/Dlubal-Software/Dlubal_CSharp_Client/assets/87471266/3f1808e7-6d32-41d2-b9f4-5a39829975e7)

-	Weird thing is that in web API tool (I used Boomerang) the data are different
-	Here is the screenshot of the response in Boomerang:

![image](https://github.com/Dlubal-Software/Dlubal_CSharp_Client/assets/87471266/e1e0014a-cbba-4fc8-b64c-dfbbdd2aa359)

-	We tried to upgrade our project to the latest Nuget (version 6.5.5) but it didn’t help
-	The same spot in the code, when RFEM 6.06.0001 is used looks like this during debugging:

![image](https://github.com/Dlubal-Software/Dlubal_CSharp_Client/assets/87471266/849bdbac-0bab-40e3-a964-284ec20a16bb)
![image](https://github.com/Dlubal-Software/Dlubal_CSharp_Client/assets/87471266/b2b7e0d4-cedb-4025-be05-9392f5e71877)

Can you help us with bringing clarity to issue 1) and help us resolve issue 2)?

Files:
[XMLs.zip](https://github.com/Dlubal-Software/Dlubal_CSharp_Client/files/15405663/XMLs.zip)

Thank you!
Štěpán

**Comment by Rebecca-Coding21** [comment, marked as answer]
Hi @sonkaste1,
Regarding point 1: The meaning of the flags is described in this [FAQ](https://www.dlubal.com/en/support-and-learning/support/faq/005529). These flags can be combined so M | I means mesh node and imperfection.

Regarding issue 2: I think you are using an old version of our webservice in Visual Studio. The currently available version is 6.6.9. This also explains the differences between Boomerang and VS. Boomerang always uses the corresponding webservice version to the RFEM version. I also tested get_member with VS and the current NuGet package and couldn't find any issues. So please update the NuGet package, that should solve your problem.
